src/linking/linking.cpp

```diff
@@ -143,7 +143,7 @@ void linkingt::detailed_conflict_report_rec(
     if(depth>0 &&
        !base_type_eq(t1.subtype(), t2.subtype(), ns))
     {
-      conflict_path=dereference_exprt(conflict_path);
+      conflict_path=dereference_exprt(conflict_path, t1.subtype());
```

Comment:
clang-format would prefer spaces around `=`.

Reply:
Also assigning it the subtype of the first expression seems a bit random. Conditioning it works as well:
```
      if(conflict_path.type().id() == ID_pointer)
        conflict_path = dereference_exprt(conflict_path);
```

Reply:
I agree with your comments / suggestions and have updated the commit accordingly. I also added a regression test case for this defect.